### PR TITLE
Feature/Multilang Support (German, Italian, Spanish, Portuguese)

### PR DESCRIPTION
**Pull request by dav009**
Connects to idio/json-wikipedia#28
Connects to idio/content-services#629
## In general:

- Added support for DE, ES, IT and PT:
- Added a script which generates the locales for potentially any language.
  `python localegen.py --lang de --o german.locale`
   
 Please be aware that such locale will not contain the following fields:
  - `list` and `disambiguation`.  
  - for the languages above I used the `disambiguation` keywords collected [here]( https://github.com/dbpedia/extraction-framework/blob/master/core/src/main/scala/org/dbpedia/extraction/wikiparser/impl/wikipedia/Disambiguation.scala)

- There is no such keywords for `list` but there is a directive for them (same as for disambiguation).

- The way in which jsonpedia marks an article as either: `list` or `disambiguation` is very weak. I added : https://github.com/idio/json-wikipedia/issues/31.

## In detail

-  The locale now has a list of NE which are language dependent, this list of NE are filtered when processing a certain language. This NE corresponds to spaces like `User`( `Benutzer` in german, `Usuario` in spanish and so on). Those NE comes from the Wikipedia API

- There is a way to check whether a page is disambiguation using a "Wikipedia Directive". So every article being a disambiguation has something like `{{Disambiguation}}` somewhere in its content(changes across languages). This directive is not included in the XML dump. And therefore we are a bit blind at the moment.

For the given languages :
- Checked Redirects
- Checked NEs
- Checked images
- Checked Categories
- Checked Basic Disambiguation
- Checked for links in other languages
- Checked for potential empty links
- Looked for utf-8 weird problems (characters being treated as ??)

### CODE REVIEW

**Comment by keynmol** [comment]
Could do with a better PR name (mention languages by full name for example), in case we need to find it.

File: localegen.py [hunk 1]
```diff
@@ -0,0 +1,128 @@
+# locale generator for jsonwikipedia
```
Comment (Lugrin) [inline localegen.py]: Can you add a shebang?

File: src/main/scala/it/cnr/isti/hpc/wikipedia/parallel/ParallelJsonpediaParser.scala [hunk 2]
```diff
@@ -8,16 +8,27 @@ import org.apache.commons.io.FileUtils
...
 object ParallelJsonpediaParser{
 
+  def getHeader(pathToSingleWikiDump: String): String ={
+    // First line contains the header:
+    // <mediawiki xmlns="http://www.mediawiki.org/xml/export-0.10/"....
+    scala.io.Source.fromFile(pathToSingleWikiDump).getLines().next()
+  }
+
+  def getNamespaces(pathToSingleWikiDump: String): String ={
+    val firstLines = scala.io.Source.fromFile(pathToSingleWikiDump).getLines().slice(0, 90000).toList
```
Comment (Lugrin) [inline src/main/scala/it/cnr/isti/hpc/wikipedia/parallel/ParallelJsonpediaParser.scala]: Would it be possible to use `par` and not assume "</siteinfo>" is within the first 90000 lines?

File: src/test/java/it/cnr/isti/hpc/wikipedia/article/pt/PortugueseArticleTest.java [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+		for (ParagraphWithLinks p : a.getParagraphsWithLinks()) {
+			for(Link l:p.getLinks()){
+				uris.add(l.getId());
+			}
+		}
+
+		assertThat(anchors, hasItems("L'Arc~en~Ciel", "The Brilliant Green", "Anime Grand Prix", "Hajime Yatate"));
+		assertThat(uris, hasItems("Sunrise", "L'Arc~en~Ciel", "The_Brilliant_Green", "Hajime_Yatate", "Mamoru_Miyano" ));
+	    assertFalse(anchors.contains("japaneselang"));
```
Comment (Lugrin) [inline src/test/java/it/cnr/isti/hpc/wikipedia/article/pt/PortugueseArticleTest.java]: tabs and spaces on this line

File: README.md [hunk 4]
```diff
@@ -152,6 +152,17 @@ and import the project in your new maven project adding the dependency:
...
 ```
 
+#### Supporting a new Language ####
+
+Supporting a new language requires creating a new locale file with specific language information.
+Most of a locale creation process is automated:
+
+1. Generate a new locale file by using `localegen.py`. As an example for generating a locale for German: `python localegen.py --lang de --o locale-de.properties`
+2. Move that locale to `resources/lang`
+3. Go to [Disambiguation Keywords](https://github.com/dbpedia/extraction-framework/blob/master/core/src/main/scala/org/dbpedia/extraction/wikiparser/impl/wikipedia/Disambiguation.scala) and look for your language keywords
```
Comment (keynmol) [inline README.md]: How about just including a static python file with those mappings? Automate even more.

https://gist.github.com/keynmol/69f28b584d4f451a77e3

File: src/main/scala/it/cnr/isti/hpc/wikipedia/parallel/ParallelJsonpediaParser.scala (hunk 2 above)
Comment (dav009) [inline src/main/scala/it/cnr/isti/hpc/wikipedia/parallel/ParallelJsonpediaParser.scala]: actually, just got the answer I wanted: 
http://stackoverflow.com/questions/34316979/wikipedia-xml-dump-where-to-get-translations-for-disambiguation-directives

File: README.md (hunk 4 above)
Comment (dav009) [inline README.md]: actually, just got the answer I wanted: http://stackoverflow.com/questions/34316979/wikipedia-xml-dump-where-to-get-translations-for-disambiguation-directives

**Comment by dav009** [comment]
@Lugrin @keynmol re re:

- Generated the locale files for major languages
- Now the script uses the locale in the dict generated by @keynmol from the java file. The disambiguation keywords for some languages i.e: FR look strange. I will leave this disambiguation keywords issue there and probably address this in a more sensible way in: https://github.com/idio/json-wikipedia/issues/31
- Following @Lugrin suggestion, I changed the way in which the namespace header is looked up

**Pull request merged**